src/bin/stratisd.rs

```diff
@@ -62,6 +87,14 @@ fn run() -> StratisResult<()> {
         .init()
         .expect("This is the first and only initialization of the logger; it must succeed");
 
+    // Setup udev events for block devices, we need to do this before we call initialize, so that
```

Comment:
Please be more precise about the bug that is avoided here. It's obvious that either you must set up the udev context before you set up the engine _or_ that this code should be placed after the engine setup code. The comment tells me that the ordering of these blocks is _not_ random, i.e., that the person who wrote the comments believed that there was actually a reason to set up the udev context before setting up the engine. This is somewhat helpful. But it still leaves me completely at a loss as to what might go wrong if the order were reversed. In order to understand the code I want to know this. I could experiment by reversing the order and seeing what happens, but that is time consuming. I'ld rather get a definite explanation right here.